src/common/task.h

```diff
@@ -228,6 +228,46 @@ int64_t task_args_add_val(task_spec *spec, uint8_t *data, int64_t length);
 object_id task_return(task_spec *spec, int64_t return_index);
 
 /**
+ * Indices into resource vectors.
+ * A resource vector maps a resource index to the number
+ * of units of that resource required.
+ *
+ * The total length of the resource vector is NUM_RESOURCE_INDICES.
+ */
+typedef enum {
+  /* Index for number of cpus the task requires. */
+  CPU_RESOURCE_INDEX = 0,
+  /* Index for number of gpus the task requires. */
+  GPU_RESOURCE_INDEX,
+  /* Total number of different resources in the system. */
+  MAX_RESOURCE_INDEX
+} resource_vector_index;
+
+#define DEFAULT_NUM_CPUS 16
```

Comment:
I'm not sure if we should have defaults. Maybe the arguments should be required to be passed in to the local scheduler?